### ISSUE TITLE
How to switch between ".h" and ".cpp"?

### ISSUE THREAD

**Issue by zhuzelong** [Yohannfra/Vim-Goto-Header#2]
In my project, the source file are `*.h` and `*.cpp`. How to switch between them?

**Comment by Yohannfra** [comment]
It was not possible before this commit, I didn't think about that use case.
 
Now it's fix, as the README says just add this line to your .vimrc to be able to switch between ".cpp" and ".h"
```vim
let g:goto_header_associate_cpp_h = 1
```

**Comment by another-pjohnson** [comment]
@Yohannfra Unfortunately you would need to do something similar to support ".h" -> ".cpp". Some projects use cc instead of cpp.

**Comment by rafaelrojasmiliani** [comment]
@another-pjohnson is right. This issue should not be closed before fixing this. I think that it would be possible to change [this](https://github.com/Yohannfra/Vim-Goto-Header/blob/247694ede014ed4326a44998b05d49cb14b91b7c/autoload/goto_header.vim#L168) string by a regular expression.

**Comment by Yohannfra** [comment]
You are right, thanks for reminding me of this issue. What I did with the **goto_header_associate_cpp_h** variable was a quick hack but it is not a proper solution. There are people using **.h**, **.hh**, **.hpp**, **.h++** and many more. I can't hard code a solution like this.

A few months ago I started to write a solution to this problem in this repository : [https://github.com/Yohannfra/Vim-Quick-Switch](https://github.com/Yohannfra/Vim-Quick-Switch) but I still have a lot of modifications not commited on my computer and didn't have the time to finish it back then. I'll try to do it asap and merge it back in this plugin.

Thanks for your interest and contribution.

**Comment by Yohannfra** [comment]
I will merge #2 for now since it's a fix for your use case but this is still a hacky solution that will be removed when I'll merge the new algorithm.
